src/textual/css/_style_properties.py

```diff
@@ -808,14 +810,47 @@ def __get__(
     ) -> Color:
         """Get a ``Color``.
 
+        If the property is "color" and the styles have a "color_auto", this value will be dynamically calculated
+        by taking into account the background colors of the attached DOMNode's ancestors.
+
         Args:
             obj (Styles): The ``Styles`` object.
             objtype (type[Styles]): The ``Styles`` class.
 
         Returns:
             Color: The Color
         """
-        return cast(Color, obj.get_rule(self.name, self._default_color))
+        if self.name != "color" or not obj.color_auto:
```

Comment:
Rather than overload the functionality of this object based on its name. We could define a property specifically for the foreground color.

This would make it easier to keep the functionality distinct. 

I think this may already be an issue. For instance, if you do `styles.background = "auto 90%"` I'm guessing that would set the foreground to auto? Which would not be desirable.